sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java

```diff
@@ -761,7 +767,16 @@ synchronized void checkPendingLogs(@NonNull GroupState groupState) {
             triggerIngestion(groupState);
         } else if (pendingLogCount > 0 && !groupState.mScheduled) {
             groupState.mScheduled = true;
-            mAppCenterHandler.postDelayed(groupState.mRunnable, groupState.mBatchTimeInterval);
+            long delay = groupState.mBatchTimeInterval;
+            if (groupState.mBatchTimeInterval > MINIMUM_TRANSMISSION_INTERVAL_IN_SECONDS) {
+                long oldestLogTime = mPersistence.getOldestLogTime(groupState.mName);
```

Comment:
That's not what was specified, sorry if there was a confusion.
Oldest logs can be discarded so this is not reliable implementation.
You could change max storage size and send a lot of logs, then kill the app before 1 day timer is due, and we would never send any logs.

To avoid this corner case, the proposal in the story was to store the timestamp when timer started as a separate value (in shared preferences for example).

Also that will be much simpler code overall as no need to modify database code.